src/puppetlabs/puppetdb/cli/import.clj

```diff
@@ -102,11 +105,20 @@
            construct-base-url
            utils/validate-cli-base-url!)))))
 
+(defn import!
+  [infile process-tar-entry-fn]
+  (let [command-versions (:command_versions (parse-metadata infile))]
+    (with-open [tar-reader (archive/tarball-reader infile)]
+      (doseq [tar-entry (archive/all-entries tar-reader)]
+        (process-tar-entry-fn tar-reader tar-entry command-versions)))))
+
 (defn -main
   [& args]
   (let [{:keys [infile base-url]} (validate-cli! args)
-        command-versions (:command_versions (parse-metadata infile))]
+        base-url-str (utils/base-url->str base-url)]
     ;; TODO: do we need to deal with SSL or can we assume this only works over a plaintext port?
-    (with-open [tar-reader (archive/tarball-reader infile)]
-      (doseq [tar-entry (archive/all-entries tar-reader)]
-        (process-tar-entry tar-reader tar-entry base-url command-versions)))))
+    (clj-http.client/post (str base-url-str "/upload")
```

Comment:
in the pdb-client namespace we alias http.client as http-client. Might help to use that here so it doesn't get overlooked in the future if someone starts using an alias for it.